/* Copyright 2020 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

import "fedlearner_webconsole/proto/common.proto";

package fedlearner_webconsole.proto;


message Participant {
    string name = 1;
    string url = 2;
    string sender_auth_token = 3;
    string receiver_auth_token = 4;
    map<string, Variable> variables = 5;
}

message Federation {
    string federation_name = 1;
    string self_name = 2;
    map<string, Participant> participants = 3;
    map<string, Variable> variables = 4;
}
